Game/Source/GDKTestGyms/Public/TestGymsReplicationGraph.h

```diff
@@ -0,0 +1,155 @@
+
+	TMap<FName, FActorRepListRefView> AlwaysRelevantStreamingLevelActors;
+
+	void PrintRepNodePolicies();
+
+private:
+
+	EClassRepNodeMapping GetMappingPolicy(UClass* Class);
+
+	bool IsSpatialized(EClassRepNodeMapping Mapping) const { return Mapping >= EClassRepNodeMapping::Spatialize_Static; }
+
+	TClassMap<EClassRepNodeMapping> ClassRepNodePolicies;
+};
+
+UCLASS()
+class UTestGymsReplicationGraphNode_AlwaysRelevant_ForConnection : public UReplicationGraphNode
+{
+	GENERATED_BODY()
+
+public:
+
+	virtual void NotifyAddNetworkActor(const FNewReplicatedActorInfo& Actor) override { }
+	virtual bool NotifyRemoveNetworkActor(const FNewReplicatedActorInfo& ActorInfo, bool bWarnIfNotFound=true) override { return false; }
+	virtual void NotifyResetAllNetworkActors() override { }
+
+	virtual void GatherActorListsForConnection(const FConnectionGatherActorListParameters& Params) override;
+
+	virtual void LogNode(FReplicationGraphDebugInfo& DebugInfo, const FString& NodeName) const override;
+
+	void OnClientLevelVisibilityAdd(FName LevelName, UWorld* StreamingWorld);
+	void OnClientLevelVisibilityRemove(FName LevelName);
+
+	void ResetGameWorldState();
+
+private:
+
+	TArray<FName, TInlineAllocator<64> > AlwaysRelevantStreamingLevelsNeedingReplication;
+
+	FActorRepListRefView ReplicationActorList;
+
+	UPROPERTY()
+	AActor* LastPawn = nullptr;
+
+	/** List of previously (or currently if nothing changed last tick) focused actor data per connection */
+	UPROPERTY()
+	TArray<FAlwaysRelevantActorInfo> PastRelevantActors;
+
+	bool bInitializedPlayerState = false;
+};
+
+/** This is a specialized node for handling PlayerState replication in a frequency limited fashion. It tracks all player states but only returns a subset of them to the replication driver each frame. */
+UCLASS()
+class UTestGymsReplicationGraphNode_PlayerStateFrequencyLimiter : public UReplicationGraphNode
+{
+	GENERATED_BODY()
+
+	UTestGymsReplicationGraphNode_PlayerStateFrequencyLimiter();
+
+	virtual void NotifyAddNetworkActor(const FNewReplicatedActorInfo& Actor) override { }
+	virtual bool NotifyRemoveNetworkActor(const FNewReplicatedActorInfo& ActorInfo, bool bWarnIfNotFound=true) override { return false; }
+
+	virtual void GatherActorListsForConnection(const FConnectionGatherActorListParameters& Params) override;
+
+	virtual void PrepareForReplication() override;
+
+	virtual void LogNode(FReplicationGraphDebugInfo& DebugInfo, const FString& NodeName) const override;
+
+	/** How many actors we want to return to the replication driver per frame. Will not suppress ForceNetUpdate. */
+	int32 TargetActorsPerFrame = 2;
+
+private:
+	
+	TArray<FActorRepListRefView> ReplicationActorLists;
+	FActorRepListRefView ForceNetUpdateReplicationActorList;
+};
+
+/** This is a specialized node only used in Spatial to ensure that connection views and targets are replicated each frame. */
+UCLASS()
+class UTestGymsReplicationGraphNode_GlobalViewTarget : public UReplicationGraphNode
```

Comment:
@improbable-danny @mironec This is a new node that has been added just for spatial